x-pack/plugins/enterprise_search/public/applications/app_search/components/multi_input_rows/multi_input_rows.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef } from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import { EuiButton, EuiButtonEmpty, EuiSpacer } from '@elastic/eui';
+
+import { CONTINUE_BUTTON_LABEL } from '../../../shared/constants';
+
+import {
+  ADD_VALUE_BUTTON_LABEL,
+  DELETE_VALUE_BUTTON_LABEL,
+  INPUT_ROW_PLACEHOLDER,
+} from './constants';
+import { InputRow } from './input_row';
+import { MultiInputRowsLogic } from './multi_input_rows_logic';
+import { filterEmptyValues } from './utils';
+
+interface Props {
+  id?: string;
+  values: string[];
+  onSubmit?(values: string[]): void;
+  onChange?(values: string[]): void;
+  submitButtonText?: string;
+  addRowText?: string;
+  deleteRowLabel?: string;
+  inputPlaceholder?: string;
+}
+
+export const MultiInputRows: React.FC<Props> = ({
+  id,
+  values: initialValues,
+  onSubmit,
+  onChange,
+  submitButtonText = CONTINUE_BUTTON_LABEL,
+  addRowText = ADD_VALUE_BUTTON_LABEL,
+  deleteRowLabel = DELETE_VALUE_BUTTON_LABEL,
+  inputPlaceholder = INPUT_ROW_PLACEHOLDER,
+}) => {
+  const logic = MultiInputRowsLogic({ id, values: initialValues });
+  console.log(logic); // key is the same despite ID/props changing???
+  const { values, hasEmptyValues, hasOnlyOneValue } = useValues(logic);
+  const { addValue, editValue, deleteValue } = useActions(logic);
+
+  const hasChanged = useRef(false);
+  useEffect(() => {
+    if (onChange && hasChanged.current) {
+      onChange(filterEmptyValues(values));
+    }
+    hasChanged.current = true;
+  }, [values]);
```

Comment:
I'm wondering if I can ditch the reference actually, now that I've changed this to a simple `onChange` fn. The onChange would fire once on mount with the same value as initialValues, which is _probably_ fine for most components sending an `onChange` instead of an onSubmit. Thoughts?

Reply:
I'm going to go ahead and remove the useRef for now. It doesn't appear to impact the Relevance Tuning page which is the main instance using `onChange`, and Curations & Synonyms use `onSubmit` and shouldn't have to worry about it.

If an instance of us using this component comes up in the future where we absolutely can't have an `onChange` event firing on component mount, then we can come back and add it in - but for now, IMO, it's unnecessary for our use-case.

Reply:
https://github.com/elastic/kibana/pull/96881/commits/3931eb36ea2498a634084f5f7dbad85dd1f55640